eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs

```diff
@@ -390,13 +393,18 @@ protected override MethodProvider[] BuildMethods()
             // Only generate tag methods if the resource model has tag properties
             if (_shouldGenerateTagMethods)
             {
+                if (updateMethodProvider is null)
```

Comment:
instead of throwing here, should we check update operation exists for _shouldGenerateTagMethods?

Reply:
the implementation of tag methods relies on update method, but whether to generate it or not I think it's separate logic. So I think check if update method exists here is fine.

Reply:
If there is no update method, how could you update tags? I don't think it's user-friendly to throw on this, it just confuses the user. And also, I don't think we should always take logic exactly from autorest.csharp.